File: java/src/main/java/com/scottlogic/domain/Book.java
package com.scottlogic.domain;

import com.fasterxml.jackson.annotation.JsonManagedReference;

import javax.persistence.*;
import java.util.ArrayList;
import java.util.Collections;
import java.util.List;

@Entity
public class Book {

    private Long id;

    private String title;
    private String author;
    private String publishDate;
    private String isbn;

    private List<Reservation> reservations;

    public Book() {
        //empty constructor required by hibernate
    }

    public Book(final String title, final String author, final String publishDate, final String isbn) {
        this.title = title;
        this.author = author;
        this.publishDate = publishDate;
        this.isbn = isbn;
        this.reservations = new ArrayList<Reservation>();
    }

    @Id
    @GeneratedValue(strategy = GenerationType.AUTO)
    public Long getId() {
        return id;
    }

    public void setId(final Long id) {
        this.id = id;
    }

    @OneToMany (mappedBy = "book")
    @JsonManagedReference
    public List<Reservation> getReservations() {
        return reservations;
    }

    public void setReservations(List<Reservation> reservations) {
        this.reservations = reservations;
    }

    public String getTitle() {
        return title;
    }

    public void setTitle(String title) {
        this.title = title;
    }

    public String getAuthor() {
        return author;
    }

    public void setAuthor(String author) {
        this.author = author;
    }

    public String getPublishDate() {
        return publishDate;
    }

    public void setPublishDate(String publishDate) {
        this.publishDate = publishDate;
    }

    public String getIsbn() {
        return isbn;
    }

    public void setIsbn(String isbn) {
        this.isbn = isbn;
    }

    @Override
    public boolean equals(final Object o) {
        if (this == o) return true;
        if (o == null || getClass() != o.getClass()) return false;

        final Book book = (Book) o;

        return id.equals(book.id);
    }

    @Override
    public int hashCode() {
        return id.hashCode();
    }
}

File: dotnet/library-grad-project/Models/Book.cs
using System.Collections.Generic;

namespace LibraryGradProject.Models
{
    public class Book
    {
        public int id { get; set; }
        public string isbn { get; set; }
        public string title { get; set; }
        public string author { get; set; }
        public string publishDate { get; set; }

        public virtual ICollection<Reservation> reservations { get; set; }

        public override bool Equals(object obj)
        {
            // If parameter is null return false.
            if (obj == null)
            {
                return false;
            }

            // If parameter cannot be cast to book return false.
            Book p = obj as Book;
            if ((System.Object)p == null)
            {
                return false;
            }

            // Return true if the fields match:
            return
                (title == p.title) &&
                (author == p.author) && 
                (isbn == p.isbn)&& 
                (publishDate == p.publishDate);
        }
    }
}